### PR TITLE
Add ability to toggle between list and progress bar

### PR DESCRIPTION
**Pull request by padietl**

### CODE REVIEW

File: app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesPresenter.java [hunk 1]
```diff
@@ -1,32 +1,49 @@
...
-        mItems = items;
-        mListView.notifyDataSetChanged();
+      public void onBeansLoaded(List<Bean> beans) {
+        ds.getSearchableItems(beans.stream().map(b -> b.getUuid()).collect(Collectors.toList()), new SearchableItemDataSource.LoadSearchableItemsCallback() {
+          @Override
+          public void onSearchableItemsLoaded(List<SearchableItem> items) {
+            mItems = items;
+            mListView.notifyDataSetChanged();
+            mListView.hideProgressBarShowList();
+          }
```
Comment (padietl) [inline app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesPresenter.java]: Give us something to do that will take time

Comment (padietl) [reply app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesPresenter.java]: Can't-do view accesses in the constructor, wait until start(). Views have not yet been initialized.

File: app/src/main/java/net/jmhossler/roastd/viewtask/SearchableItemListFragment.java [hunk 2]
```diff
@@ -70,8 +74,20 @@ public void navigateToShop(String shopId) {
...
+
+  @Override
+  public void hideProgressBarShowList() {
+    mProgressBar.setVisibility(View.GONE);
+    mListRecyclerView.setVisibility(View.VISIBLE);
+  }
+
   /// Holds the View of each item in the RecyclerView
-  class SearchableItemHolder extends RecyclerView.ViewHolder implements
+  static class SearchableItemHolder extends RecyclerView.ViewHolder implements
```
Comment (padietl) [inline app/src/main/java/net/jmhossler/roastd/viewtask/SearchableItemListFragment.java]: Non-static inner classes have implicit references to their outer class. Static classes break this link and thus help prevent activity/fragment memory leaks due to rogue references to them.

File: app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesPresenter.java (hunk 1 above)
Comment (jmhossler) [reply app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesPresenter.java]: well, damn. This was where I was doing work. I'll need these changes, I think, to continue what I was doing. I think I have most of the logic for getting and retrieving the user favorites, though

Comment (padietl) [reply app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesPresenter.java]: My changes here other than moving the query into main are purely for demoing the progress bar. I can revert this file

**Pull request merged**